src/track/beats.h

```diff
@@ -1,45 +1,75 @@
-#ifndef BEATS_H
-#define BEATS_H
+#pragma once
 
-#include <QObject>
-#include <QString>
-#include <QList>
 #include <QByteArray>
+#include <QList>
+#include <QObject>
 #include <QSharedPointer>
+#include <QString>
 
+#include "proto/beats.pb.h"
+#include "util/logger.h"
 #include "util/memory.h"
 #include "util/types.h"
 
-namespace {
-    double kMaxBpm = 500;
-}
-
+namespace mixxx {
 class Beats;
-typedef QSharedPointer<Beats> BeatsPointer;
+using BeatsPointer = std::shared_ptr<Beats>;
+using BeatList = QList<track::io::Beat>;
+using SamplePos = double;
+using FrameNum = double;
```

Comment:
We prefer typedef here. The using syntax for typdef is useful in a template context.

Reply:
No problem. In other projects they prefer using and me too :)
May be we can have a specific mention in the coding guidelines?
Done.

Reply:
I think we need actual classes here, not simple `using` or `typedef`. Then we can [delete the implicit conversions to and from `double`](https://foonathan.net/2015/10/overload-resolution-1/) to prevent confusion and errors.

Reply:
`FrameNum` is a strange name. "Num" doesn't add anything that "Frame" doesn't convey.

Reply:
On further thought, let's not add a `SamplePos` class. All code that would use that should move to using the new Frame class.